### PR TITLE
fix: add strict checking to getRegisteredCommandIndex function

### PR DESCRIPTION
**Pull request by allroundexperts**
This PR fixes the `getRegisteredCommandIndex` function that it does a strict matching of the input shortcut with the list of available commands. If the strict matching fails, only then it checks if there is a loose match (without modifier key).

Ref:
https://github.com/Expensify/App/issues/18480#issuecomment-1537541205

### CODE REVIEW

File: src/KeyCommand/index.js [hunk 1]
```diff
@@ -111,17 +111,25 @@ function getRegisteredCommandIndex(json) {
...
-        matchesEnter(item) && matchesModifierFlags(item)
-    ) || matchesEscape(item)
-        || matchesEnter(item)
-        || matchesUpArrow(item)
-        || matchesDownArrow(item)
-        || matchesLeftDown(item)
-        || matchesRightDown(item));
+    // This does a strict check first to see if there is a match
+    // https://github.com/Expensify/App/issues/18480
+    const strictIndex = _.findIndex(commands, item => (
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/KeyCommand/index.js:116]: I have no idea, first time seeing this repository 😂

[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/KeyCommand/index.js:116]: @allroundexperts  any updates on this?

[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/KeyCommand/index.js:116]: thanks this is good to begin with. Appreciate it :)

**Review by mananjadhav — APPROVED** [review]
LGTM. @mountiny Can you look at this PR?

We can complete the checklist on the App PR.

**Comment by mountiny** [comment]
1.0.1 version

**Pull request merged**